src/functor.jl

```diff
@@ -29,6 +29,35 @@ macro functor(args...)
   functorm(args...)
 end
 
+function makeflexiblefunctor(m::Module, T, pfield)
+  pfield = QuoteNode(pfield)
+  @eval m begin
+    $Functors.functor(::Type{<:$T}, x) = begin
```

Comment:
```suggestion
    function $Functors.functor(::Type{<:$T}, x)
       pfields = x.$pfield
```
Using `getproperty` instead of `getfield` also leaves the option to override it and add more dynamism to what fields are included (e.g. derived from values of the current fields). My unscientific benchmarking seemed to indicate that hoisting the fields outside of the closure incurs a couple extra allocations, but is ~3x faster.